app/controllers/cases_controller.rb

```diff
@@ -81,6 +82,25 @@ def all_activity
     @activities = CaseActivity.all
   end
 
+  def rename_file
+
+    @case = Case.find(params[:id])
+    file = @case.files.find{params["file_id"]}
```

Comment:
this doesn't return the file with that id and was causing it to always update the first file's name (if you add 2 files and rename the second, it updates the first). Just need parens instead:

```
    file = @case.files.find(params["file_id"]})
```